### PR TITLE
Add version to hw sets and script files

### LINKED ISSUE #443: Add `version` field to hardware set config files
This would be useful if we ever want to make non-backwards-compatible changes to the file format.

### LINKED ISSUE #444: Add `version` field to measure script files
As with #443, it would be useful if measure script files had a version field in case we ever want to make non-backwards-compatible changes to the format.

### PR DESCRIPTION
**Pull request by dalonsoa**
# Description

Second take at this issue, leaving the version number just for the schema wihout actually saving it in any object. 

Fixes #443 
Fixes #444

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Documentation (non-breaking change that adds or improves the documentation)
- [ ] Optimisation (non-breaking, back-end change that speeds up the code)

## Key checklist

- [x] Pre-commit hooks run successfully (`pre-commit run -a`)
- [x] All tests pass (`pytest`)
- [x] The documentation builds without warnings (`mkdocs build -s`)
- [ ] Check the GUI still works (if relevant)
- [ ] Check the code works with actual hardware (if relevant)
- [ ] Check the `pyinstaller`-built executable works (if relevant)

## Further checks

- [x] Code is commented, particularly in hard-to-understand areas
- [x] Tests have been added or an issue has been opened to tackle that in the future. (Indicate issue here: # (issue))

### CODE REVIEW

File: finesse/gui/hardware_set/hardware_set.py [hunk 1]
```diff
@@ -82,6 +85,7 @@ def _non_empty(x: Any) -> bool:
 
 _hw_set_schema = Schema(
     {
+        "version": Const(1, "Version number must be 1"),
```
Comment (alexdewar) [inline finesse/gui/hardware_set/hardware_set.py:88]:
```suggestion
        "version": Const(CURRENT_HW_SET_VERSION, "Version number must be 1"),
```

**Review by alexdewar — APPROVED** [review]
One small suggestion, but otherwise LGTM!

**Comment by dalonsoa** [comment]
@alexdewar , I don't know how to make Codacy happy. Either I'm not setting the config file right, or it is ignoring that setting. Or it is just not working at all.

**Comment by alexdewar** [comment]
@dalonsoa Dw about it -- there various other things in the codebase that Codacy isn't happy about either. Tbh I'm not sure it adds anything over the linters we already have, so maybe we should just remove it.

**Pull request merged**